GroupByVariable: Add support for default values

mdvictor opened the pull request:
Adds support for dashboard level default values in the groupBy variable.

SImilar to AdHoc filters, these new values saved on the dashboard for the groupBy represent default values that one can revert back to. Editing the values will show a restore icon and on click will restore the groupBy to the dashboard level default values.

TODO:
- [x] tests
- [x] UI polishing
<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>6.13.0--canary.1122.15186594626.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/scenes-react@6.13.0--canary.1122.15186594626.0
  npm install @grafana/scenes@6.13.0--canary.1122.15186594626.0
  # or 
  yarn add @grafana/scenes-react@6.13.0--canary.1122.15186594626.0
  yarn add @grafana/scenes@6.13.0--canary.1122.15186594626.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

torkelo commented on packages/scenes/src/variables/groupby/GroupByVariable.tsx:
```diff
@@ -154,6 +157,39 @@ export class GroupByVariable extends MultiValueVariable<GroupByVariableState> {
+  }
+
+  private _activationHandler = () => {
+    if (this.state.defaultValues && this.checkIfRestorable(this.state.value)) {
+      this.setState({ restorable: true });
+    }
```
not sure I understand this, looks like restorable is just calculated state (result of current value != default values). 

feels like this could be a simple useMemo function in the rendering of the group by

mdvictor replied:
changed to useMemo, thank you!

bfmatei reviewed (approved):
lgtm

mdvictor commented on packages/scenes/src/variables/groupby/GroupByVariable.tsx:
```diff
@@ -28,6 +29,7 @@ export interface GroupByVariableState extends MultiValueVariableState {
   baseFilters?: AdHocVariableFilter[];
   /** Datasource to use for getTagKeys and also controls which scene queries the group by should apply to */
   datasource: DataSourceRef | null;
+  defaultValues?: { text: VariableValue; value: VariableValue };
```
Agree, have moved to singular naming. Also added a comment explaining the field. Thank you!

mdvictor commented on packages/scenes/src/variables/groupby/GroupByVariable.tsx:
```diff
@@ -154,6 +156,41 @@ export class GroupByVariable extends MultiValueVariable<GroupByVariableState> {
+    }
+  };
+
+  public checkIfRestorable(values: VariableValue) {
```
Done, thank you!

mdvictor commented:
I refactored this a bit as it was failing some specific use-cases. We cannot use default flags on the values as we do not know what values were previously (e.g. if dashboard one has 2 default values and we remove one we still have one default and the next dashboard doesn't know that there were 2 before. It will see the one and think that those are all the previous dashboards default values, thus not setting it as restorable, when it should) 

I've reverted to my initial idea of having a restorable flag in the url for the groupBy, this works much better and accommodates all usecases, although we need the restorable property back in the state to be able to make it work properly with urlSync across dashboard changes and activation/reactivations of the groupBy.

I've tested this in both using the reload dashboard feature flag and without and works properly in both situations.
